### ISSUE TITLE
Dashboard with ChartRangeFilter unexpectedly filters leading and trailing null rows

### ISSUE THREAD

**Issue by mirkosessa** [google/google-visualization-issues#2967]
Hi everybody,
not sure this is a bug or a feature.

I need to build a Dashboard consisting of a TableChart and a ChartRangeFilter.
The input DataTable may contain columns with leading and/or trailing null values.
![image](https://user-images.githubusercontent.com/43070597/186860623-ffc57194-581b-4fa0-be4b-eb1c504d8c94.png)

The unexpected behavior is the following:
the Dashboard omits all the leading and/or trailing rows of the input DataTable where all row values are null.
![image](https://user-images.githubusercontent.com/43070597/186860917-9b1dba8e-2d79-4cbc-b0ff-9d4d88604523.png)

Additionally, when the ChartRangeFilter UI is restricted to show only some columns of the DataTable (i.e. using  `chartView: {columns: [0, 1]}`), the same behavior happens, that is, the Dashboard omits all the leading and/or trailing rows of the input DataTable that contain null values in the columns specified with the ChartView option.
![image](https://user-images.githubusercontent.com/43070597/186861092-a446c1d4-af7f-4ab1-a108-8a276960410d.png)

As far as I understood, this is an unexpected behavior.
I would expect to see all the rows of the input DataTable, null-rows included, since I'm not explicitly performing any selection/filtering on rows.
If this is a feature, is there any way of keeping leading and/or trailing null rows in the DashBoard?

Here the fiddle:
[https://jsfiddle.net/msessa/p5ejqf1m/24](https://jsfiddle.net/msessa/p5ejqf1m/24)

Thank you in advance.

**Comment by dlaliberte** [comment]
I agree that filtering out leading and trailing rows with null values  is unexpected behavior.  It's been this way for many versions, however, so I hesitate to change it without considering the implications.  I believe there is actually a dependency on this behavior related to data that comes from a Google spreadsheet, especially when sorting the table based on the data in a column that might contain null values (or empty cells).

Maybe the solution will have to involve adding an option, analogous to the interpolateNulls option, which will avoid filtering out these rows.

**Comment by mirkosessa** [comment]
I rarely use Google spreadsheet as source, but I see the potential backward compatibility issue.

Adding a new option, with the proper default value, seems a good solution to avoid regression.
The new option might have three values:

1. `'none'`: all rows of the DataTable are shown, regardless of the use of the  `chartView: {columns: [...]}` option in the ChartRangeFilter control;
2. `'filterGlobal'`: all leading and trailing null-rows are filtered at the level of the entire DataTable, ignoring the specific columns shown in the ChartRangeFilter (i.e. those specified by the option `chartView: {columns: [...]}`);
3. `'filterView'`: all leading and trailing null-rows are filtered considering only the specific columns shown in the ChartRangeFilter (i.e. those specified by the option `chartView: {columns: [...]}`. This third option value would be the current behavior.

BTW, a similar null-rows filtering feature might be of interest also for other standalone charts (charts that are not part of a DashBoard), possibly as a general functionality of the DataTable/DataView object.

HTH.
